src/Contracts/Makeable.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Contracts;
+
+interface Makeable
+{
+    /**
+     * Instantiate a new class with the arguments.
+     *
+     * @param mixed ...$arguments
+     * @return static
```

Comment:
@juse-less why are you returning static here but in ArrayStore you uses $this?

Reply:
Depending on what you're referring to -

The `make()` method(s) returns a new instance, whereas calls to objects returns _the same_ instance.
`static` is the same class, where `$this` is the very same instance you call the method on.

The native type hint is still `static`, however, because PHP doesn't support type hinting `$this`.

```php
class MyClass
{
    public static function make(): static
    {
        return new static;
    }

    public function setValue(mixed $value): static
    {
        // ... set value, etc ...

        return $this;
    }
}
```

Reply:
I'm guessing you're referring to this method (as an example), and the others like it:
https://github.com/Sammyjo20/Saloon/blob/7cb97128732c6e1f7cf9c75f20e173922e3005d7/src/Contracts/ArrayStore.php#L29

The reason is that the method returns the same instance, and not a new instance.
If, for example, the class would be immutable, we'd return `new static` instead of `$this`, and therefore also type hint with `@return static`.

Reply:
Got it! Thank your for your explanation and your contribution!